File: ui/src/api/history.ts
import request, {Response} from "@/lib/request.ts";

export interface History {
    id?: number;
    taskId: number;
    name: string;
    gitHash: string;
    userId: number;
    userName: string;
    createdAt?: string;
}


export const getHistoryByTaskId = (taskId: number) => {
    return request<Response<History[]>>({
        url: "/v1/histories",
        method: "get",
        params: {taskId}
    });
}

export const restoreHistory = (id: number) => {
    return request<Response<number>>({
        url: `/v1/history/restore/${id}`,
        method: "post"
    });
}

File: internal/models/history.go
package models

// History 历史版本模型
type History struct {
	Model
	TaskID   uint   `json:"taskId"`
	Name     string `json:"name"`
	GitHash  string `json:"gitHash"`
	UserID   uint   `json:"userId"`
	UserName string `json:"userName"`
}

// TableName 指定表名
func (History) TableName() string {
	return "histories"
}
